Sources/GRPC/GRPCServerCodec.swift

```diff
@@ -75,8 +78,9 @@ extension GRPCServerCodec: ChannelOutboundHandler {
 
     case .message(let message):
       do {
-        let messageData = try message.serializedData()
-        context.write(self.wrapOutboundOut(.message(messageData)), promise: promise)
+        var responseBuffer = context.channel.allocator.buffer(capacity: protobufMetadataSize)
```

Comment:
I think you can just allocate an empty buffer here, the `writer` reserves the space we need.